model-optimizer/extensions/back/ReverseInputChannels.py

```diff
@@ -295,38 +295,41 @@ class ReverseChannelsPropagationUp(BackReplacementPattern):
         'Subtract': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
         'Pow': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
         'Convert': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_eltwise(node, rc),
-
-        'Pad': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through(node, rc),
+        'Pad': lambda node, rc: ReverseChannelsPropagationUp.lift_up_through_pad(node, rc),
     }
 
     @staticmethod
-    def lift_up_through(node: Node, reverse_channels: Node):
+    def lift_up_through_pad(node: Node, reverse_channels: Node):
         r"""
         BEFORE                       AFTER
 
                                      previous_op
                                           \
         previous_op  previous_op       ReverseChannels  previous_op
                  \     /                           \     /
-                   Node                             Node
+                   Pad                              Pad
                     |                                |
               ReverseChannels                      next_op
                     |
                  next_op
 
-        returns boolean value whatever we should continue propagating current ReverseChannels operation up or not
+        returns two objects:
+        first - boolean value whatever we should continue propagating current ReverseChannels operation up or not
+        second - list of ReverseChannels operations that were produced while propagating reverse_channels up
         """
         if node.is_in_port_connected(0):
             node_input_port_0 = node.in_port(0)
-            reverse_channels_out_npde = reverse_channels.out_port(0).get_connection().get_destination().node
+            reverse_channels_out_nodes = reverse_channels.out_port(0).get_connection().get_destinations()
             reverse_channels.out_port(0).disconnect()
-
+            reverse_channels.in_port(0).disconnect()
             src = node_input_port_0.get_connection().get_source()
             node_input_port_0.get_connection().set_source(reverse_channels.out_port(0))
             src.connect(reverse_channels.in_port(0))
-            node.out_port(0).get_connection().set_destination(reverse_channels_out_npde.in_port(0))
-            return True
-        return False
+            for reverse_channels_destination in reverse_channels_out_nodes:
+                node.out_port(0).get_connection().add_destination(reverse_channels_destination)
```

Comment:
I would recommend to rewrite this code like this:

```python
            pad_producer_port = node.in_port(0).get_connection().get_source()
            node.in_port(0).disconnect()
            # remove connection between Pad and ReverseChannels (Pad may have side consumers)
            reverse_channels.in_port(0).get_connection().remove()
            reverse_channels.in_port(0).connect(pad_producer_port)
            reverse_channels.out_port(0).get_connection().set_source(node.out_port(0))
            node.in_port(0).connect(reverse_channels.out_port(0))
```

Anyway, I still see the Split/Concat sub-graph in the original model mentioned in the ticket 58776. So looks like the solution is not full or wrong.

Reply:
@iimironov, please check that it fixes a bug from 58776.

Reply:
![image](https://user-images.githubusercontent.com/29801542/132472327-1336dfe7-79a7-416b-bbd9-083293952630.png)
On the screen result topology before and after fix of 58776 
The main idea of fix bug is move ReverseChannels after Pad layer. For model text-detection-00004 it works.
Also we have following case:
![image](https://user-images.githubusercontent.com/29801542/132473391-c7ca4671-b6e2-465d-843f-df78c5fee1dd.png)
In the case we cannot move down ReverseChannels after Pad layer because elemtwise have tow outputs.

And also we have model from 60934
In the case we cannot fuse ReverseChannels with some one op befor or after Pad layer.